File: ex94.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def inorderTraversal(self, root):
        """
        :type root: TreeNode
        :rtype: List[int]
        """
        
        global r
        r = []
    
        self.inorder(root)
        
        return r
            
            
    def inorder(self, p):
        global r
        
        if p != None:
            self.inorder(p.left)
            
            r.append(p.val)
            
            self.inorder(p.right)
        
        

File: code/ex94.c
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     struct TreeNode *left;
 *     struct TreeNode *right;
 * };
 */
/**
 * Return an array of size *returnSize.
 * Note: The returned array must be malloced, assume caller calls free().
 */
int* inorderTraversal(struct TreeNode* root, int* returnSize) {

    TreeNode* p;
    p = root;

    if(p != NULL){
        inorderTraversal(p->left, returnSize);

        printf("%d\n", p->val);

        inorderTraversal(p->right, returnSize);
    }
    
}